src/ObjWriting/ObjContainer/IWD/IWDWriter.cpp

```diff
@@ -0,0 +1,106 @@
+#include "IWDWriter.h"
+
+#include "../minizip/zip.h"
+
+#include <algorithm>
+#include <iostream>
+
+class IWDWriterImpl final : public IWDWriter
+{
+public:
+    explicit IWDWriterImpl(const std::string& iwdFileName, ISearchPath* assetSearchPath)
+        : m_filename(iwdFileName),
+          m_asset_search_path(assetSearchPath),
+          m_files(0)
+    {
+        m_zipfile = zipOpen(m_filename.c_str(), APPEND_STATUS_CREATE);
+        if (!m_zipfile)
+        {
+            throw std::runtime_error("Error creating IWD file: \"" + m_filename + "\"");
+        }
+    }
+
+    void AddFile(std::string fileName) override
+    {
+        m_files.emplace_back(std::move(fileName));

```

Comment:
This is not true. The `std::move` does make the `emplace_back` use the move constructor. That is necessary because calling the method itself does make the copy already since it's not a reference.

So it's correct to either use `std::move` here or otherwise make the parameter a `const std::string&` and use no `std::move`